src/components/TodoItem.tsx/TodoItem.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useState } from 'react';
+import cn from 'classnames';
+import { useTodos } from '../../TodosContext';
+import { Todo } from '../../types/types';
+
+export const TodoItem: React.FC<{
+  todo: Todo;
+  onToggle: () => void;
+  onDelete: () => void;
+}> = ({ todo, onToggle, onDelete }) => {
+  const [editing, setEditing] = useState(false);
+  const [editedTitle, setEditedTitle] = useState(todo.title);
+
+  const { setTodos } = useTodos();
+
+  const handleDoubleClick = () => {
+    setEditing(true);
+  };
+
+  const handleSave = () => {
+    if (editedTitle.trim() === '') {
```

Comment:
```suggestion
    if (!editedTitle.trim()) {
```

can be a bit simplify